mslib/mscolab/file_manager.py

```diff
@@ -778,3 +779,63 @@ def import_permissions(self, import_op_id, current_op_id, u_id):
         except IntegrityError:
             db.session.rollback()
             return False, None, "Some error occurred! Could not import permissions. Please try again."
+
+    def save_view_settings(self, op_id, user, view_settings):
+        """Save view settings for an operation and user to the database."""
+
+        if not self.is_member(user.id, op_id):
+            logging.error("User %s is not a member of operation %s", user.id, op_id)
+            return False, "Access denied: User is not a member of this operation"
+        try:
+            settings_str = json.dumps(view_settings)
+            logging.info(f"settings_str : {settings_str}")
+            view_setting = ViewSettings.query.filter_by(u_id=user.id, op_id=op_id).first()
+            logging.info(f"User: {user.id}, Operation: {op_id}, View settings: {view_settings}")
+            if view_setting:
+                view_setting.settings = settings_str
+                view_setting.updated_at = datetime.datetime.now(tz=datetime.timezone.utc)
+                db.session.commit()
+                logging.info("Updated view settings for user %s, operation %s", user.id, op_id)
+                return True, "View settings updated successfully"
+            else:
+                view_setting = ViewSettings(op_id=op_id, u_id=user.id, settings=settings_str)
+                db.session.add(view_setting)
+                db.session.commit()
+                logging.info("View settings saved for user %s, operation %s", user.id, op_id)
+                return True, "View settings saved successfully"
+        except Exception as e:
+            db.session.rollback()
+            logging.error("Error saving view settings for user %s, operation %s: %s", user.id, op_id, str(e))
+            return False, f"Failed to save view settings: {str(e)}"
+
+    def get_view_settings(self, op_id, user):
+        """Retrieve view settings for an operation and user from the database."""
+        if not self.is_member(user.id, op_id):
+            logging.error("User %s is not a member of operation %s", user.id, op_id)
+            return False, "Access denied: User is not a member of this operation", {}
+
+        try:
+            settings_exist = db.session.query(ViewSettings).filter_by(op_id=op_id).first() is not None
+            if not settings_exist:
+                return True, "No settings found for this operation", {"views": [], "global": {}}
```

Comment:
What is the reason for this preliminary check, when you are just interested in the settings for (op_id, user)?